wgpu-core/src/command/bundle.rs

```diff
@@ -869,20 +884,34 @@ struct State {
 }
 
 impl State {
-    fn vertex_limits(&self) -> (u32, u32) {
-        let mut vertex_limit = !0;
-        let mut instance_limit = !0;
-        for vbs in &self.vertex {
+    fn vertex_limits(&self) -> VertexLimitState {
+        let mut vert_state = VertexLimitState {
+            vertex_limit: u32::MAX,
+            vertex_limit_slot: 0,
+            instance_limit: u32::MAX,
+            instance_limit_slot: 0,
+        };
+        for (idx, vbs) in self.vertex.iter().enumerate() {
             if vbs.stride == 0 {
                 continue;
             }
             let limit = ((vbs.range.end - vbs.range.start) / vbs.stride) as u32;
             match vbs.rate {
-                wgt::InputStepMode::Vertex => vertex_limit = vertex_limit.min(limit),
-                wgt::InputStepMode::Instance => instance_limit = instance_limit.min(limit),
+                wgt::InputStepMode::Vertex => {
+                    vert_state.vertex_limit = vert_state.vertex_limit.min(limit);
+                    if vert_state.vertex_limit == limit {
```

Comment:
nit: it seems like this would be cleaner to just do with a single `if` instead of `if` + `min`:
```
if limit < vert_state.vertex_limit {
  vert_state.vertex_limit = limit;
  vert_state.vertex_limit_slot = idx as _;
}
```